lib/pengine/utils.c

```diff
@@ -2034,139 +2034,240 @@ append_versioned_params(xmlNode *versioned_params, const char *ra_version, xmlNo
     }
     g_hash_table_destroy(hash);
 }
+
+static void
+append_all_versioned_params(pe_resource_t *rsc, pe_node_t *node,
+                            pe_action_t *action, pe_working_set_t *data_set)
+{
+    const char *ra_version = NULL;
+    xmlNode *local_versioned_params = NULL;
+    pe_rsc_action_details_t *details = pe_rsc_action_details(action);
+
+    local_versioned_params = create_xml_node(NULL, XML_TAG_RSC_VER_ATTRS);
+    pe_get_versioned_attributes(local_versioned_params, rsc, node, data_set);
+    if (xml_op != NULL) {
+        ra_version = crm_element_value(xml_op, XML_ATTR_RA_VERSION);
```

Comment:
where is xml_op supposed to come from - don't see it static here - probably should be passed from the caller?

Reply:
Whoops, yes. I haven't bother compiling the versioned attributes code since it's got major problems already. I may end up dropping that code, but there are a few significant issues it uncovered that do need to be fixed, so I wanted to get at least that far before deciding.